pkg/controllers/dynakube/extension/eec/reconciler.go

```diff
@@ -70,3 +84,104 @@ func (r *reconciler) Reconcile(ctx context.Context) error {
 
 	return r.createOrUpdateStatefulset(ctx)
 }
+
+func (r *reconciler) reconcileTLSSecret(ctx context.Context) error {
+	query := k8ssecret.Query(r.client, r.client, log)
+
+	secret, err := query.Get(ctx, types.NamespacedName{
+		Name:      getTLSSecretName(r.dk.Name),
+		Namespace: r.dk.Namespace,
+	})
+
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return err
+	}
+
+	if r.dk.Spec.Templates.ExtensionExecutionController.TlsRefName != "" {
```

Comment:
I still don't understand this check, here we introduced new method `GetExtensionsTlsSecretName `, but on line 100 we check only if user set custom TlsRefName, what about default one? and on line 92 we use alternative method?

Reply:
We still support if the user sets `Spec.Templates.ExtensionExecutionController.TlsRefName`. Hence, here I check if the user has set it, and if so, I delete whatever existing self-signed cert was created before.

We only create the self-signed certificate if the user doesn't set it themselves.

Reply:
yeah make sense.